pkg/utils/options/options.go

```diff
@@ -41,20 +42,24 @@ func MustParse() Options {
 
 // Options for running this binary
 type Options struct {
-	ClusterName     string
-	ClusterEndpoint string
-	MetricsPort     int
-	HealthProbePort int
-	WebhookPort     int
-	KubeClientQPS   int
-	KubeClientBurst int
+	ClusterName            string
+	ClusterEndpoint        string
+	MetricsPort            int
+	HealthProbePort        int
+	WebhookPort            int
+	KubeClientQPS          int
+	KubeClientBurst        int
+	DefaultInstanceProfile string
 }
 
 func (o Options) Validate() (err error) {
 	err = multierr.Append(err, o.validateEndpoint())
 	if o.ClusterName == "" {
 		err = multierr.Append(err, fmt.Errorf("CLUSTER_NAME is required"))
 	}
+	if o.DefaultInstanceProfile == "" {
```

Comment:
This is not required right? IIUC, It's required to be passed as a CLI arg OR provided in the provisioner. 

Or am I mistaken and we're requiring a default one passed in the CLI arg?

Reply:
That was something we talked about earlier in the review. I had it originally as you mentioned, with the default at the controller being optional. We changed it to an override. But yeah... that does affect it's coupling to a specific cloud provider. 

We don't have any mechanism to validate the flags based on provider, but maybe that's an idea too? For AWS-based deploys, it's required?

Reply:
I could see a use-case of provisioners where they are handed out per team but a central team manages the Karpenter install. As the central team, I don't know if I'd want to put a default InstanceProfile in the controller for all provisioners to inherit as a default value. Seems too easy to make a mistake and have a provisioner inherit a broader set of permissions than desired. You could make cases for using IRSA and applying best practices of only doing a least privileges profile in the controller options, but it won't always happen. 

Maybe a default instanceProfile doesn't exist for a use-case. Maybe a platform team will craft individual roles for each provisioner and if the argument is required on the controller, they now have to specify a dummy profile that wouldn't work with a node. 

If DefaultInstanceProfile is optional on the controller, the webhook defaulting logic could validate the provisioner based on both values where the provisioner one wins since it's more specific. The webhook should have all of that knowledge at validation time.

Reply:
I agree with the sentiment and had started with that idea, having the default as optional. The tricky part is that there's no guarantee the webhook and controller are configured the same way, since they are deployed separately (granted the Helm chart will make sure they are configured similarly). But, there's a chance the webhook doesn't know a default that the controller might be using. 

I'm good with the idea of reverting it back (as it makes it less backwards-disruptive for those wanting to update). Just want to make sure everyone is good with this approach then. Cc/ @ellistarn

Reply:
To resolve and document the resolution to this discussion, all three of us chatted on Slack and determined the best approach going forward is to make the setting of the instance profile optional at both the API and CLI. We will then try and fail early in the runtime reconciliation. With this approach, we don't have to worry about defaults coming and going, but also have the flexibility for various use cases.